File: NetFlexAPI/TestProject1/Subscription.cs
using System;

namespace TestProject;

public class Subscription
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Price { get; set; }
    
    public Subscription(Guid id, string name,string price)
    {
        Id = id;
        Name = name;
        Price = price;
    }
}

File: NetFlexAPI/GiraffeAPI/Models/Subscription.fs
module GiraffeAPI.Models.Subscription

open System
open System.ComponentModel.DataAnnotations
open Microsoft.EntityFrameworkCore.Metadata.Internal
open Ply.TplPrimitives

[<CLIMutable>]  
type Subscription =
    {
        Id : Guid
        Name : string
        Price : string
    }

    member this.HasErrors =
        if this.Name = null || this.Name = "" then Some "Name is required"
        elif this.Price = null || this.Price = "" then Some "Cost is required"
        else None
        